schemas/task.py

```diff
@@ -46,19 +48,26 @@ class _TaskBase(SQLModel):
             Dictionary (saved as JSON) of the default parameters of the task
     """
 
-    name: str
-    source: str
-    
-    
+    name: str = Field(alias="name")
+    source: str = Field(alias="source")
+
+
 class TaskUpdate(_TaskBase):
-    name: Optional[str]  # type:ignore
-    input_type: Optional[str]  # type:ignore
-    output_type: Optional[str]  # type:ignore
-    command: Optional[str]  # type:ignore
-    source: Optional[str]  # type:ignore
+    name: Optional[str]
+    input_type: Optional[str]
+    output_type: Optional[str]
+    command: Optional[str]
+    source: Optional[str]
     default_args: Optional[Dict[str, Any]]  # type:ignore
     meta: Optional[Dict[str, Any]]  # type:ignore
 
+    @root_validator(pre=True)
+    def not_empty_strings(cls, values):
+        for k, v in values.items():
+            if isinstance(v, str):
+                values[k] = validate_str(v, k)
```

Comment:
This is not a validation check, but an actual change to the attribute.
For the moment I would rather stick with operations that do not modify the attribute, but rather "only" fail if it is not valid.

If we decide we want to be "smart" and modify the user's input, this should be a general decision - and we should probably do it in several places. At the moment, I propose we don't follow that path.

Reply:
What should we do if the user enters this input `name="   name            "`?
Raise a `ValueError` or set `name="name"`?

Reply:
As per our last discussion: OK, let's strip the starting/ending whitespaces